src/main/webapp/app/exercises/quiz/manage/quiz-question-list-edit-existing.component.ts

```diff
@@ -170,16 +171,58 @@ export class QuizQuestionListEditExistingComponent implements OnChanges {
         if (!this.importFile) {
             return;
         }
+
+        const fileName = this.importFile.name;
+        const fileExtension = fileName.split('.').last()?.toLowerCase();
+
+        if (fileExtension === 'zip') {
+            await this.handleZipFile(this.importFile);
+        } else {
+            this.handleJsonFile(this.importFile);
+        }
+    }
+
+    handleJsonFile(file: File) {
         const fileReader = this.generateFileReader();
         fileReader.onload = () => this.onFileLoadImport(fileReader);
-        fileReader.readAsText(this.importFile);
+        fileReader.readAsText(file);
     }
 
-    async onFileLoadImport(fileReader: FileReader) {
+    async handleZipFile(file: File) {
+        const jszip = new JSZip();
+
+        try {
+            const zipContent = await jszip.loadAsync(file);
+            const jsonFiles = Object.keys(zipContent.files).filter((fileName) => fileName.endsWith('.json'));
+
+            const images = await this.extractImagesFromZip(zipContent);
+            const jsonFile = zipContent.files[jsonFiles[0]];
+            const jsonContent = await jsonFile.async('string');
+            await this.processJsonContent(jsonContent, images);
+        } catch (error) {
+            return;
+        }
+    }
+    async extractImagesFromZip(zipContent: JSZip) {
+        const images: Map<string, File> = new Map();
+
+        for (const [fileName, zipEntry] of Object.entries(zipContent.files)) {
+            if (fileName.endsWith('.png') || fileName.endsWith('.jpg') || fileName.endsWith('.jpeg')) {
```

Comment:
Please make this a constant array of supported file extensions and use a comment here and on the server to remind developers of keeping these in sync.

Reply:
This statement was unneccessary of me to include now that i think about it. I'm adjusting the import function to retain the original extension correctly instead of converting it to PNG. Since the upload from Zip is handled as a regular upload, I will remove this if statement and let the original logic check.